background.js

```diff
@@ -1,41 +1,30 @@
 function conditions() {
-  var conditionList = []
-  var request = new XMLHttpRequest();
-  request.open('GET', 'https://raw.githubusercontent.com/vaibhavmule/ycinfo/master/ycstartup.json', true);
-
-  request.onload = function() {
-    if (request.status >= 200 && request.status < 400) {
-      // Success!
-      var ycStartups = JSON.parse(request.responseText);
-      Object.keys(ycStartups).forEach(function (key) {
-        conditionList.push(
-          new chrome.declarativeContent.PageStateMatcher({
-            pageUrl: { urlMatches:  key + '\\..*' }
-          })
-        )
-      })
-    } else {
-      // We reached our target server, but it returned an error
-
+  return fetch(`https://raw.githubusercontent.com/vaibhavmule/ycinfo/master/ycstartup.json`)
+    .then(function(res) {
+    if (res.status === 200) {
+      return res.json();
     }
-  };
-
-  // request.onerror = function() {
-  //   // There was a connection error of some sort
-  // };
-
-  request.send();
-
-  return conditionList;
+    })
+    .then(function(ycStartups) {
+    console.log(ycStartups);
+    return Object.keys(ycStartups).map(function (key) {
+      return new chrome.declarativeContent.PageStateMatcher({
+        pageUrl: { urlMatches:  key + '\\..*a' }
+      })
+    });
+    })
 }
-
+ 
 chrome.runtime.onInstalled.addListener(function(details) {
   chrome.declarativeContent.onPageChanged.removeRules(undefined, function() {
-    chrome.declarativeContent.onPageChanged.addRules([
-      {
-        conditions: conditions(),
-        actions: [ new chrome.declarativeContent.ShowPageAction()]
-      }
-    ]);
+    conditions().then(function(res) {
```

Comment:
may be chang `res` to `conditions`.
```suggestion
    conditions().then(function(conditions) {
```